api/models.py

```diff
@@ -17,6 +17,55 @@ class Trip(models.Model):
     def __str__(self):
         return self.name
 
+    def possible_dates(self):
+        possible_dates = []
+        is_date_valid = True
+        is_user_available = False
+
+        id = self.users.first().id
+
+        user_one_trip_users = self.tripuser_set.filter(user_id=id)
+
+        all_users = self.users.all()
+
+        if len(all_users) > 1:
+            all_users = list(all_users)
+        else:
+            all_users = [all_users]
+
+        other_users = []
+        for user in all_users:
+            if user not in other_users:
+                other_users.append(user)
+
+        other_users.pop(0)
```

Comment:
i'm not sure how yet, but I definitely think this section is a great target for a refactor later on, either moving it to a helper method or just shortening it if we can